app/forms/policy_form.rb

```diff
@@ -80,6 +81,8 @@ def update(attrs)
   end
 
   def save
+    return unless self.policy.valid? || valid?
```

Comment:
This seems a little off to me. If we're using a form object, all of the validations we care about on creating/updating a policy should be found in this object - not split between the form and the Policy model. Is there any particular reason why all Policy validations can't simply be contained within this class, thus simplifying the above to `unless valid?`. 

(As aside - you don't need `self` here. It's only really required when writing values.)